sdk/lib/atl/atlcoll.h

```diff
@@ -639,6 +642,34 @@ POSITION CAtlList<E, ETraits>::AddTail(INARGTYPE element)
     return (POSITION)Node;
 }
 
+template <typename E, class ETraits>
+void CAtlList<E, ETraits>::AddHeadList(_In_ const CAtlList<E, ETraits>* plNew)
+{
+    ATLASSERT(plNew != NULL && plNew != this);
+    
+    POSITION pos = plNew->GetTailPosition();
+    
+    while (pos)
+    {
+        INARGTYPE element = plNew->GetPrev(pos);
+        AddHead(element);
+    }
+}
+
+template <typename E, class ETraits>
+void CAtlList<E, ETraits>::AddTailList(_In_ const CAtlList<E, ETraits>* plNew)
+{
+    ATLASSERT(plNew != NULL && plNew != this);
+    
+    POSITION pos = plNew->GetHeadPosition();
+    
+    while (pos)
+    {
+        INARGTYPE element = plNew->GetNext(pos);
+        AddTail(element);
```

Comment:
Is it possible to do
```suggestion
        AddTail(plNew->GetNext(pos));
```
or the C++ magic needs the intermediate variable?
(+ same question for the other function)

Reply:
Works totally fine even in my one-line curly braces-less style.